### PR TITLE
Migrate available dependencies to `ksp`

### PR DESCRIPTION
**Pull request by wzieba**
### Description

This PR replaces usage of `kapt` with `ksp` for this dependencies, where it's possible.

Additionally it migrates the project to use Room Gradle Plugin instead of passing arguments via annotation processor arguments.

### Testing

It's a build-related work: green CI checks should be just fine. If not, smoke test of an app can be also performed.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: libs/login/build.gradle [hunk 1]
```diff
@@ -1,16 +1,16 @@
 plugins {
     id "com.android.library"
     id "org.jetbrains.kotlin.android"
```
Comment (ParaskP7) [inline libs/login/build.gradle]:
```suggestion
    alias(libs.plugins.android.library)
    alias(libs.plugins.kotlin.android)
```

File: libs/login/build.gradle [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    id "org.jetbrains.kotlin.kapt"
     alias(libs.plugins.kotlinx.kover)
+    alias(libs.plugins.ksp)
 }
 
 android {
     namespace "org.wordpress.android.login"
 
-    compileSdkVersion rootProject.compileSdkVersion
```
Comment (ParaskP7) [inline libs/login/build.gradle]: Minor (🔍): Consider removing the extra empty line.

**Review by ParaskP7 — APPROVED** [review]
👋 @wzieba !

I have reviewed and tested this PR as per the instructions, everything works as expected, great job! 🌟 🌟

PS: We are now left to only migrate the `wellsql.processor` dependency and the `libs:fluxc-processor` module, and then, that's it. Easier said than done, I know, but still, this here is good progress. We've now isolated this migration on the `fluxc` module only! 🥇 

-----

I have left one suggestion (💡) and minor (🔍) comment for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

**Pull request merged**